### PR TITLE
removing megam reader and writer

### LINKED ISSUE #532: Remove MegaM Reader and Writer classes?
Is anyone using the `MegaMReader` or `MegaMWriter` classes in `skll`? If not, is there any reason to maintain these? Is there any code that still uses the megam classifier?

@aoifecahill  @desilinguist @mulhod @bndgyawali?

### PR DESCRIPTION
**Pull request by bndgyawali**
This PR closes #532. 

Removing megam reader and writer.
Updating comments, and documents mentioning megam
Updating test cases that are using megam file types.

### CODE REVIEW

**Comment by bndgyawali** [comment]
`pep8speaks` not working?

**Comment by bndgyawali** [comment]
I do not know what the `codecov` is doing here. When I ran tests locally, this is what i see

`$MPLBACKEND=Agg nosetests -v --with-coverage --cover-package skll -c=.coveragerc tests/test_*py`
```
Name                                              Stmts   Miss  Cover
---------------------------------------------------------------------
skll/__init__.py                                     10      0   100%
skll/config.py                                      313     15    95%
skll/data/__init__.py                                 7      0   100%
skll/data/dict_vectorizer.py                          4      0   100%
skll/data/featureset.py                             158     14    91%
skll/data/readers.py                                294     29    90%
skll/data/writers.py                                184     11    94%
skll/experiments.py                                 603     27    96%
skll/learner.py                                     794     32    96%
skll/logutils.py                                     30      0   100%
skll/metrics.py                                      70      2    97%
skll/utilities/__init__.py                            0      0   100%
skll/utilities/compute_eval_from_predictions.py      71      2    97%
skll/utilities/filter_features.py                    63      1    98%
skll/utilities/generate_predictions.py               93      2    98%
skll/utilities/join_features.py                      54      1    98%
skll/utilities/plot_learning_curves.py               26      2    92%
skll/utilities/print_model_weights.py                59      3    95%
skll/utilities/run_experiment.py                     31      1    97%
skll/utilities/skll_convert.py                       72      5    93%
skll/utilities/summarize_results.py                  18      1    94%
skll/version.py                                       3      0   100%
---------------------------------------------------------------------
TOTAL                                              2957    148    95%
----------------------------------------------------------------------
Ran 4505 tests in 1100.602s

OK
```

File: doc/utilities.rst [hunk 1]
```diff
@@ -131,7 +131,7 @@ Positional Arguments
 
 .. option:: input_file(s)
 
-    One or more csv file(s), jsonlines file(s), or megam file(s) (with or without the
+    One or more csv file(s), or jsonlines file(s) (with or without the
```
Comment (mulhod) [inline doc/utilities.rst]: Nitpick/question: Get rid of the comma if there are only two things. But, also, why are there only two things? That is, shouldn't we mention all the file types supported here (or link to a section that enumerates them)? It seems strange to focus on CSV and jsonlines exclusively here. Also, I suggest changing "csv" and "jsonlines" to "\`\`.csv\`\`" and "\`\`.jsonlines\`\`" as they are used elsewhere (same applies to additional files added).

**Comment by bndgyawali** [comment]
Looks it is ready for review, (or merging). Will address any comments further.

**Comment by desilinguist** [comment]
@bndgyawali looks like there are some conflicts that need to be resolved based on changes we made for v2.1? Can you please update. Thanks!

File: skll/data/writers.py [hunk 2]
```diff
@@ -762,7 +708,7 @@ def _sanitize(name):
 
     def _write_line(self, id_, label_, feat_dict, output_file):
         """
-        Write the current line in the file in MegaM format.
+        Write the current line in the file in given format.
```
Comment (desilinguist) [inline skll/data/writers.py:711]: To match the other writers, this should be:

```suggestion
        Write the current line in the file in this Writer's format.
```

File: skll/utils/commandline/generate_predictions.py [hunk 3]
```diff
@@ -42,8 +42,8 @@ def main(argv=None):
                         help='Model file to load and use for generating '
                              'predictions.')
     parser.add_argument('input_files',
-                        help='A space-separated list of CSV, TSV, or '
-                             'jsonlines files (with or without the label '
+                        help='A space-separated list of csv file, or json file, '
+                             '(with or without the label '
```
Comment (desilinguist) [inline skll/utils/commandline/generate_predictions.py:46]: Why are we changing this?

Comment (bndgyawali) [reply skll/utils/commandline/generate_predictions.py:46]: not sure how it got changed. i will revert it back.

**Comment by bndgyawali** [comment]
> A suggestion and a question. Very minor stuff.

I addressed them.

**Pull request merged**